dev/com.ibm.ws.microprofile.metrics.2.0/src/com/ibm/ws/microprofile/metrics/helper/PrometheusBuilder.java

```diff
@@ -36,30 +37,43 @@
     private static final String QUANTILE = "quantile";
 
     @FFDCIgnore({ IllegalStateException.class })
-    public static void buildGauge(StringBuilder builder, String name, Gauge<?> gauge, String description, Double conversionFactor, String tags, String appendUnit) {
-        // Skip non number values
-        Number gaugeValNumber = null;
-        Object gaugeValue = null;
-        try {
-            gaugeValue = gauge.getValue();
-        } catch (IllegalStateException e) {
-            // The forwarding gauge is likely unloaded. A warning has already been emitted
-            return;
-        }
-        if (!Number.class.isInstance(gaugeValue)) {
-            Tr.event(tc, "Skipping Prometheus output for Gauge: " + name + " of type " + gauge.getValue().getClass());
-            return;
-        }
-        gaugeValNumber = (Number) gaugeValue;
-        if (!(Double.isNaN(conversionFactor))) {
-            gaugeValNumber = gaugeValNumber.doubleValue() * conversionFactor;
-        }
+
+    public static void buildGauge(StringBuilder builder, String name, String description, Map<MetricID, Metric> currentMetricMap, Double conversionFactor,
+                                  String appendUnit) {
         getPromTypeLine(builder, name, "gauge", appendUnit);
         getPromHelpLine(builder, name, description, appendUnit);
-        getPromValueLine(builder, name, gaugeValNumber, tags, appendUnit);
+
+        for (MetricID mid : currentMetricMap.keySet()) {
+            // Skip non number values
+            Number gaugeValNumber = null;
+            Object gaugeValue = null;
+            try {
+                gaugeValue = ((Gauge) currentMetricMap.get(mid)).getValue();
+            } catch (IllegalStateException e) {
+                // The forwarding gauge is likely unloaded. A warning has already been emitted
+                return;
+            }
+            if (!Number.class.isInstance(gaugeValue)) {
+                Tr.event(tc, "Skipping Prometheus output for Gauge: " + name + " of type " + ((Gauge) currentMetricMap.get(mid)).getValue().getClass());
```

Comment:
Will this message be printed every time `/metrics` is served?  i.e. will it be printed a lot of times in the log?

Reply:
Yes they would be emitted every time, but only if they did something wrong! Or if something went horribly wrong. For example this event is emitted if the gauge doesn't track a Number value. The other events emitted in PrometheusMetricWriter will happen if the metadata's type isn't correctly set.